src/random_ray/vtk_plot.cpp

```diff
@@ -0,0 +1,154 @@
+// fission source data to .vtk file that can be directly
+// loaded and displayed by Paraview.
+void plot_3D_vtk()
+{
+  // Rename .h5 plot filename(s) to .vtk filenames
+  for (int p = 0; p < model::plots.size(); p++) {
+    PlottableInterface* plot = model::plots[p].get();
+    plot->path_plot() =
+      plot->path_plot().substr(0, plot->path_plot().find_last_of('.')) + ".vtk";
+  }
+
+  // Print header information
+  print_plot();
+
+  // Outer loop over plots
+  for (int p = 0; p < model::plots.size(); p++) {
+
+    // Get handle to OpenMC plot object and extract params
+    Plot* openmc_plot = dynamic_cast<Plot*>(model::plots[p].get());
+
+    // Random ray plots only support voxel plots
+    if (openmc_plot == nullptr) {
+      warning(fmt::format("Plot {} is invalid plot type -- only voxel plotting "
+                          "is allowed in random ray mode.",
+        p));
+      continue;
+    } else if (openmc_plot->type_ != Plot::PlotType::voxel) {
+      warning(fmt::format("Plot {} is invalid plot type -- only voxel plotting "
+                          "is allowed in random ray mode.",
+        p));
+      continue;
+    }
+
+    int Nx = openmc_plot->pixels_[0];
+    int Ny = openmc_plot->pixels_[1];
+    int Nz = openmc_plot->pixels_[2];
+    Position origin = openmc_plot->origin_;
+    Position width = openmc_plot->width_;
+    Position ll = origin - width / 2.0;
+    double x_delta = width.x / Nx;
+    double y_delta = width.y / Ny;
+    double z_delta = width.z / Nz;
+    std::string filename = openmc_plot->path_plot();
+
+    int negroups = data::mg.num_energy_groups_;
+
+    // Perform sanity checks on file size
+    uint64_t bytes = Nx * Ny * Nz * (negroups + 1 + 1 + 1) * sizeof(float);
+    write_message(5, "Processing plot {}: {}... (Estimated size is {} MB)",
+      openmc_plot->id(), filename, bytes / 1.0e6);
+    if (bytes / 1.0e9 > 1.0) {
+      warning("Voxel plot specification is very large (>1 GB). Plotting may be "
+              "slow.");
+    } else if (bytes / 1.0e9 > 100.0) {
+      fatal_error("Voxel plot specification is too large (>100 GB). Exiting.");
+    }
+
+    // Relate voxel spatial locations to random ray source regions
+    std::vector<int> voxel_indices(Nx * Ny * Nz);
+
+#pragma omp parallel for collapse(3)
+    for (int z = 0; z < Nz; z++) {
+      for (int y = 0; y < Ny; y++) {
+        for (int x = 0; x < Nx; x++) {
+          Position sample;
+          sample.z = ll.z + z_delta / 2.0 + z * z_delta;
+          sample.y = ll.y + y_delta / 2.0 + y * y_delta;
+          sample.x = ll.x + x_delta / 2.0 + x * x_delta;
+          Particle p;
+          p.r() = sample;
+          bool found = exhaustive_find_cell(p);
+          int i_cell = p.lowest_coord().cell;
+          int64_t source_region_idx =
+            random_ray::source_region_offsets[i_cell] + p.cell_instance();
+          voxel_indices[z * Ny * Nx + y * Nx + x] = source_region_idx;
```

Comment:
Oh wow! So the plotting routine isn't shared with `Plot` at all. This should definitely just be a subclass of `PlottableInterface`.